src/plugins/transform/scss.ts

```diff
@@ -46,26 +45,46 @@ export function transformScssPlugin({ options }: ModuleContext): VitePlugin {
 }
 
 function createScssTransform(fileExtension: string, sassVariables?: string | boolean): (code: string) => string {
-  const sassImportCode = ['@import \'quasar/src/css/variables.sass\'', '']
+  return (content) => {
+    // Check if content already imports Quasar variables to avoid conflicts
+    const hasQuasarVariables
+      = content.includes('quasar/src/css/variables')
+        || content.includes('quasar/dist/')
+        || (content.includes('@use') && content.includes('quasar'))
```

Comment:
This check may have false positives especially with the last expression. (Consumer may have used `@use` and `quasar` label at some comment)

For example:
```sass
@use '../assets/my-variables.sass' as *
/* quasar is awesome */
.button
  background: red
```

I think its better to release this as a breaking change rather than trying to avoid it.

Reply:
You're right, that's a tricky one to catch. We could check if contents contains `@use "quasar` or `@use 'quasar`. Do you think that would that be sufficient? Agree with the breaking change... could be a new major version with a beta release to give users a chance to test against their setup.